### ISSUE TITLE
npm run dev picks up IPV6 address and fails with listen EINVAL on Fedora

### ISSUE THREAD

**Issue by sebastien** [observablehq/framework#812]
On Fedora 39, running `npm run dev` on a freshly created blank project:

```
npm run dev

> dev
> observable preview


Unexpected error: listen EINVAL: invalid argument XXX::XXXX:XXXX:XXXX:XXXX:3000

Tip: To see the full stack trace, run with the --debug flag.

If you think this is a bug, please file an issue at
↳ https://github.com/observablehq/framework/issues
```

where the argument is my laptop's IPv6 address instead of the IPv4.

Now trying to run `npm run observablepreview --host=localhost --port=8000` also leads to the same error.

**Comment by mythmon** [comment]
@sebastien Could you run this with the `--debug` flag, to help pinpoint where the error is coming from?

Also, do you have any idea what is different about your system that is causing IPv6 there? I'm also on a Linux system, so maybe I could try and reproduce this, which would make things easier to fix.

**Comment by osgav** [comment]
I have the same issue on an earlier version of Fedora. I haven't had any luck getting a stack trace to appear by adding `--debug` – I'm not sure where `npm` commands like this expect that flag, so I tried everywhere but I got the same message (as per sebastian above) each time. 

Are there other debugging options we could try?

**Comment by mbostock** [comment]
Was this fixed by #854?

**Comment by mythmon** [comment]
> I have the same issue on an earlier version of Fedora. I haven't had any luck getting a stack trace to appear by adding `--debug` – I'm not sure where `npm` commands like this expect that flag, so I tried everywhere but I got the same message (as per sebastian above) each time.
> 
> Are there other debugging options we could try?

When using `npm run`, to pass arguments to the underlying command instead of npm, you have to separate them with `--`. The full command you would need to run is

```sh
npm run dev -- --debug
```

**Comment by elevenchars** [comment]
I'm having the same issue. Here's the output from `npm run dev -- --debug`:

```
[willi@fedora]~/code/hello-framework% npm run dev -- --debug

> dev
> observable preview --debug


Unexpected error: listen EINVAL: invalid argument fe80::xxxx:xxxx:xxxx:xxxx:xxxx
The full error follows

node:internal/errors:563
    ErrorCaptureStackTrace(err);
    ^

Error: listen EINVAL: invalid argument fe80::xxxx:xxxx:xxxx:xxxx:xxxx
    at __node_internal_captureLargerStackTrace (node:internal/errors:563:5)
    at __node_internal_uvExceptionWithHostPort (node:internal/errors:660:12)
    at Server.setupListenHandle [as _listen2] (node:net:1855:21)
    at listenInCluster (node:net:1920:12)
    at GetAddrInfoReqWrap.doListen [as callback] (node:net:2069:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:8) {
  code: 'EINVAL',
  errno: -22,
  syscall: 'listen',
  address: 'fe80::xxxx:xxxx:xxxx:xxxx',
  port: 3000
}
```

**Comment by Fil** [comment]
This upstream node issue seems relevant
https://github.com/nodejs/node/issues/51732

**Comment by mbostock** [comment]
I expect this has been fixed by #854; @elevenchars likely has a HOSTNAME environment variable set to an ipv6 address.

**Comment by cedricr** [comment]
I got the same problem with Fedora 39 with a brand new install; I can confirm that switching to 
`"@observablehq/framework": "observablehq/framework#6b1e943771dfb81e7eaf41dc96fa5e6b7be695ff",`  (that is, #854) as a dependency works, where as 
` "@observablehq/framework": "observablehq/framework#3f09fab3ea186531eddeda670b27233af1bc09ce",` (the previous commit) won’t. 
So, fixed for me!

**Comment by elevenchars** [comment]
> I expect this has been fixed by #854; @elevenchars likely has a HOSTNAME environment variable set to an ipv6 address.

My `HOSTNAME` variable is set to `fedora`. I suspect it's the upsteam issue @Fil linked since it is a link-local address.

**Comment by mbostock** [comment]
Fixed in #854.

**Comment by alabhyajindal** [comment]
Hi, I'm facing the same issue on Fedora, getting the same error as @elevenchars. 

Can you help me understand how to get it to work?
